concurrent_test.go

```diff
@@ -511,6 +537,23 @@ func (rs historyRecords) Swap(i, j int) {
 	rs[i], rs[j] = rs[j], rs[i]
 }
 
+func validateIncrementalTxid(rs historyRecords, op OperationType) error {
```

Comment:
I feel that `validateIncrementalTxid` itself should have tests for its capability of detecting non incremental txid history (also for `validateSerializable`). Do we have such tests already?

Reply:
Yes, it's a good point. Will add. thx

Reply:
Note I am planning to split the big concurrent_test.go file into a couple of small files in followup PR. I will add test cases to cover both of the functions by then.

Reply:
I think that `validateIncrementalTxid` in it's current form is just too complicated. If we separated checking txid being incremental vs checking writes increasing txid by 1 then there was no need for tests.

Overall writing tests for a test is a bad idea. Test code should be clear and simple enough that it can be read. Having each assert independent from the other should help here.